.github/workflows/eden_gcp.yml

```diff
@@ -0,0 +1,103 @@
+---
+name: EdenGCP
+on:  # yamllint disable-line rule:truthy
+  push:
+    branches: [master]
+# yamllint disable rule:line-length
+jobs:
+  integration:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        hv: ["kvm", "xen"]
+    steps:
+      - name: get eve
+        uses: actions/checkout@v2
+        with:
+          path: 'eve'
+      - name: get eden
+        uses: actions/checkout@v2
+        with:
+          repository: 'lf-edge/eden'
+      - name: fetch or build eve
+        env:
+          TAG: pr${{ github.event.pull_request.number }}
+          CACHE: evebuild/danger
+        run: |
+          if docker pull "$CACHE:$TAG-${{ matrix.hv }}"; then
+             docker tag  "$CACHE:$TAG-${{ matrix.hv }}" "lfedge/eve:$TAG-${{ matrix.hv }}"
+             docker tag  "$CACHE:$TAG-${{ matrix.hv }}" "lfedge/eve:$TAG-${{ matrix.hv }}-amd64"
+          else
+             make -C eve pkgs
+             make -C eve ROOTFS_VERSION="$TAG" HV=${{ matrix.hv }} eve
+          fi
+          echo "TAG=$TAG" >> $GITHUB_ENV
+      - name: Public IP
+        id: ip
+        uses: haythem/public-ip@v1.2
+      - name: setup packages
+        run: |
+          sudo apt update
+          sudo apt install -y qemu-utils openvpn
+          echo "$OVPN_FILE" | base64 -d > ./config.ovpn
+        env:
+          OVPN_FILE: ${{ secrets.OVPN_FILE }}
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.GCP_PROJECT_ID }}
+          service_account_key: ${{ secrets.GCP_SA_KEY }}
+          export_default_credentials: true
+      - name: set firewall & clean
+        run: |
+          gcloud compute instances delete eve-edengcp-actions-${{ matrix.hv }}-${{github.run_number}} -q --zone=us-west1-a || echo "not exists"
+          gcloud compute images delete eve-edengcp-actions-${{ matrix.hv }}-${{github.run_number}} -q || echo "not exists"
```

Comment:
So I'm confused -- what happens if this gets triggered while the previous run is still running -- wouldn't it delete the image that the other run is using?

Reply:
it's using run number.` eve-edengcp-actions-${{ matrix.hv }}-${{github.run_number}}`